### PR TITLE
notifyParentItemRangeRemoved and notifyParentItemRangeChanged

### PR DESCRIPTION
**Pull request by paul-turner**
Adds the two range methods and simplifies logic a bit for single item change/remove.

Part of resolving #88

### CODE REVIEW

**Comment by dgreenhalgh** [comment]
Code looks good, will test in the morning.

File: app/src/main/java/com/ryanbrooks/expandablerecyclerviewsample/linear/horizontal/HorizontalLinearRecyclerViewSampleActivity.java [hunk 1]
```diff
@@ -372,6 +378,53 @@ public void onClick(View v) {
         }
     };
 
+    private OnClickListener mRemoveTwoParentsClickListener = new OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            mTestDataItemList.remove(mTestDataItemList.size() - 1);
```
Comment (dgreenhalgh) [inline app/src/main/java/com/ryanbrooks/expandablerecyclerviewsample/linear/horizontal/HorizontalLinearRecyclerViewSampleActivity.java]: `ArrayOutOfBoundsException` here when removing last two parents in an empty list.

**Comment by dgreenhalgh** [comment]
Needs a re-up

**Comment by paul-turner** [comment]
Fixed

**Comment by dgreenhalgh** [comment]
Looks good :+1:

**Pull request merged**